core/src/qrcode/MQRDecoder.cpp

```diff
@@ -0,0 +1,392 @@
+{
+	// Read two characters at a time
+	std::string buffer;
+	while (count > 1) {
+		if (bits.available() < 11) {
+			return DecodeStatus::FormatError;
+		}
+		int nextTwoCharsBits = bits.readBits(11);
+		buffer += ToAlphaNumericChar(nextTwoCharsBits / 45);
+		buffer += ToAlphaNumericChar(nextTwoCharsBits % 45);
+		count -= 2;
+	}
+	if (count == 1) {
+		// special case: one character left
+		if (bits.available() < 6) {
+			return DecodeStatus::FormatError;
+		}
+		buffer += ToAlphaNumericChar(bits.readBits(6));
+	}
+	// See section 6.4.8.1, 6.4.8.2
+	if (fc1InEffect) {
+		// We need to massage the result a bit if in an FNC1 mode:
+		for (size_t i = 0; i < buffer.length(); i++) {
+			if (buffer[i] == '%') {
+				if (i < buffer.length() - 1 && buffer[i + 1] == '%') {
+					// %% is rendered as %
+					buffer.erase(i + 1);
+				} else {
+					// In alpha mode, % should be converted to FNC1 separator 0x1D
+					buffer[i] = static_cast<char>(0x1D);
+				}
+			}
+		}
+	}
+	TextDecoder::AppendLatin1(result, buffer);
+	return DecodeStatus::NoError;
+}
+
+static DecodeStatus DecodeNumericSegment(BitSource& bits, int count, std::wstring& result)
+{
+	// Read three digits at a time
+	std::string buffer;
+	while (count >= 3) {
+		// Each 10 bits encodes three digits
+		if (bits.available() < 10) {
+			return DecodeStatus::FormatError;
+		}
+		int threeDigitsBits = bits.readBits(10);
+		if (threeDigitsBits >= 1000) {
+			return DecodeStatus::FormatError;
+		}
+		buffer += ToAlphaNumericChar(threeDigitsBits / 100);
+		buffer += ToAlphaNumericChar((threeDigitsBits / 10) % 10);
+		buffer += ToAlphaNumericChar(threeDigitsBits % 10);
+		count -= 3;
+	}
+	if (count == 2) {
+		// Two digits left over to read, encoded in 7 bits
+		if (bits.available() < 7) {
+			return DecodeStatus::FormatError;
+		}
+		int twoDigitsBits = bits.readBits(7);
+		if (twoDigitsBits >= 100) {
+			return DecodeStatus::FormatError;
+		}
+		buffer += ToAlphaNumericChar(twoDigitsBits / 10);
+		buffer += ToAlphaNumericChar(twoDigitsBits % 10);
+	} else if (count == 1) {
+		// One digit left over to read
+		if (bits.available() < 4) {
+			return DecodeStatus::FormatError;
+		}
+		int digitBits = bits.readBits(4);
+		if (digitBits >= 10) {
+			return DecodeStatus::FormatError;
+		}
+		buffer += ToAlphaNumericChar(digitBits);
+	}
+
+	TextDecoder::AppendLatin1(result, buffer);
+	return DecodeStatus::NoError;
+}
+
+bool IsTerminator(const BitSource& bits, const Version& version)
+{
+	const int bitsRequired = TerminatorBitsLength(version);
+	const int bitsAvailable = std::min(bits.available(), bitsRequired);
+	return bits.peakBits(bitsAvailable) == 0;
+}
+
+/**
+ * <p>QR Codes can encode text as bits in one of several modes, and can use multiple modes
+ * in one QR Code. This method decodes the bits back into text.</p>
+ *
+ * <p>See ISO 18004:2006, 6.4.3 - 6.4.7</p>
+ */
+ZXING_EXPORT_TEST_ONLY DecoderResult DecodeBitStream(ByteArray&& bytes, const Version& version,
```

Comment:
This file is almost identical to `QRDecoder.cpp`. The few differences in this function should easily be manageable with a boolean flag discriminating between standard and micro version. This would also require to use the `QRCode::CodecMode` which should be possible (I think). I might be overlooking something and would need to actually get this code into my IDE and start merging it to see any issues.